### PR TITLE
Switch to the React PostHogProvider

### PR DESCRIPTION
**Pull request by coffeeorgreentea**
## What Changed:

I was troubleshooting posthog as I thought it was not working. Turns out I has adblock on! lol
This is a quick improvement i made while troubleshooting that, it switches to using the PosthogProvider component so that we can integrate things missed as metioned by @Knar33 here https://github.com/Oneirocom/Magick/pull/704#pullrequestreview-1400354771 easier.

## How to test:
setup your envs
```
VITE_APP_POSTHOG_ENABLED=false
VITE_APP_POSTHOG_API_KEY=key
```
and then login to posthog to check for events/recordings.

### CODE REVIEW

**Comment by Knar33** [comment]
I can't get this one to work, I'm pretty sure I have it setup correctly. I tried to set VITE_APP_POSTHOG_ENABLED to both true and "true" and tried my API key with quotes and without quotes (not sure which is right). I'm not getting the console log 'Posthog enabled'.

**Comment by Knar33** [comment]
> I can't get this one to work, I'm pretty sure I have it setup correctly. I tried to set VITE_APP_POSTHOG_ENABLED to both true and "true" and tried my API key with quotes and without quotes (not sure which is right). I'm not getting the console log 'Posthog enabled'.

I figured out why, the events only get registered if magick is running in an iframe (makes sense). I tried creating a test html page with an iframe but it seems like you can only run magick in an iframe from the trusted url. Not sure how to test this.

**Comment by coffeeorgreentea** [comment]
> > I can't get this one to work, I'm pretty sure I have it setup correctly. I tried to set VITE_APP_POSTHOG_ENABLED to both true and "true" and tried my API key with quotes and without quotes (not sure which is right). I'm not getting the console log 'Posthog enabled'.
> 
> I figured out why, the events only get registered if magick is running in an iframe (makes sense). I tried creating a test html page with an iframe but it seems like you can only run magick in an iframe from the trusted url. Not sure how to test this.

My bad I should have mentioned that in the 'what changed'! Yeah I changed it so it to that so it wont run for local users regardless of .env vars to keep it cleaner. You can test it by running the cloud repo locally and accessing a project through the iframe there

**Review by Knar33 — APPROVED** [review]
I was able to log some events to Posthog after I got the Cloud IDE running locally. The Posthog live event viewer is being really slow so I couldn't do an in-depth test but I think it's good.

File: apps/client/src/main.tsx [hunk 1]
```diff
@@ -82,16 +75,23 @@ if (window === window.parent) {
...
         const { config } = payload
-
-        // Initialize posthog
-        if (POSTHOG_ENABLED) {
-          console.log('Posthog enabled')
-          posthog.init(POSTHOG_API_KEY, {
-            api_host: 'https://app.posthog.com',
-          })
+        const Root = () => {
+          if (POSTHOG_ENABLED === 'true') {
```
[the remark this answers is not in the document]
Comment (Knar33) [reply apps/client/src/main.tsx:79]: It works like this, my issue was caused by something else.

**Pull request merged**